packnet_sfm/geometry/camera_generic.py

```diff
@@ -178,31 +178,38 @@ def _get_value_coords(x, coords):
         direction_norm = direction.view(B, 3, H*W)
         direction_norm = direction_norm / \
             torch.norm(direction_norm, dim=1, keepdim=True)
-        direction_view = direction_norm.view(B, 3, H*W).squeeze()
 
-        patch_ray_view = ray_surf_patch.squeeze().view(3, H*W, K)
+        projections = []
+        for b in range(B):
+            direction_view = direction_norm[b,:,:].view(1, 3, H*W).squeeze()
```

Comment:
This `view().squeeze()`, and potentially a few others below, while seemingly useless, is required for successful training. Not quite sure what's going on here but seems to be a sort of `deepcopy` stand-in.